Multisig/UI/UI Library/Views/TokenAmountField.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  TokenAmountField.swift
+//  Multisig
+//
+//  Created by Vitaly Katz on 20.12.21.
+//  Copyright © 2021 Gnosis Ltd. All rights reserved.
+//
+
+import UIKit
+
+class TokenAmountField: UINibView {
+    
+    @IBOutlet weak var borderImage: UIImageView!
+    @IBOutlet weak var iconImage: UIImageView!
+    @IBOutlet weak var amountTextField: UITextField!
+    @IBOutlet weak var symbolLabel: UILabel!
+    @IBOutlet weak var errorLabel: UILabel!
+    
+    var balance: String {
+        get { amountTextField.text ?? "" }
+        set { amountTextField.text = newValue }
+    }
+    
+    override func commonInit() {
+        super.commonInit()
+    }
+    
+    override func awakeFromNib() {
+        symbolLabel.setStyle(.tertiary)
+        amountTextField.setStyle(.primary)
+        amountTextField.placeholder = "Amount"
+        errorLabel.setStyle(.error)
+        errorLabel.isHidden = true
+    }
+    
+    func showToken(symbol: String, logoURL: String? = nil, amount: String = "") {
+        iconImage.setCircleShapeImage(url: URL(string: logoURL ?? ""), placeholder:  UIImage(named: "ico-token-placeholder")!)
+        symbolLabel.text = symbol
+        amountTextField.text = amount
+        borderImage.tintColor = .gray4
+        errorLabel.isHidden = true
+    }
+    
+    func showError(message: String) {
+        errorLabel.text = message
+        errorLabel.isHidden = false
+        borderImage.tintColor = .error
+    }
+}
+
+// allow only decimal numbers to be entered in the amount field
+extension TokenAmountField: UITextFieldDelegate {
+    func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
+        
+        let decimalSeparator = Locale.autoupdatingCurrent.decimalSeparator ?? "."
```

Comment:
decimal separator can be 1 of 3 different things (this is how keyboard works), please see more here: https://github.com/gnosis/SwiftCryptoTokenFormatter/blob/6f605fd7bd61327a6006919e28bf3f0632d1c556/SwiftCryptoTokenFormatter/SwiftCryptoTokenFormatter/TokenFormatter.swift#L14

Reply:
and it doesn't adhere to the locale. I killed a lot of time in the past to find out this.

Reply:
it should adhere to locale according to https://github.com/gnosis/safe/wiki/How-to-format-amounts. 
Using Locale.current.decimalSeparator

Reply:
it doesn't, there are combinations that even if locale is set to one value, the actual decimal separator that keyboard shows is something else

Reply:
try simulating different locales + regions: US, German, UK, Japanese, Korean, French ...

Reply:
I have tried different locales/regions combinations and each time I was getting the expected decimal separator. The same separator also was used in amounts displayed in the balances list.

Reply:
separator displayed in the decimalPad keyboard was adhering to locale in my tests.